src/harness/sas_test_harness.py

```diff
@@ -325,50 +358,59 @@ def generateCbsdRecords(registration_requests, grant_requests_list):
     # Check if groupingParam optional parameter present in CBSDRecord then
     # add to registration object.
     if registration_request.has_key('groupingParam'):
-      internal_data['registration'].update({'groupingParam': registration_request['groupingParam']})
+      internal_data['registration'].update({'groupingParam': 
+                         registration_request['groupingParam']})
 
     internal_data['grants'] = []
 
-    # Read all grant records if mulitiple grant records are associated with single cbsd.
+    # Read all grant records if mulitiple grant records are associated with 
+    # single cbsd.
     for grant_request in grant_requests:
       grant_data = {}
-      if grant_request.has_key('id'):
-        grant_data['id'] = grant_request['id']
-      if grant_request.has_key('operationParam'):
-        grant_data['operationParam'] = grant_request['operationParam']
-      if grant_request.has_key('channelType'):
-        grant_data['channelType'] = grant_request['channelType']
+
+      # Auto-generating GrantData 'id' field.
+      grant_data['id'] = 'SAMPLE_ID_{:05}'.format(random.randrange(1, 10 ** 5)) 
+      
+      assert grant_request.has_key('operationParam'), "operationParam does not exist in GrantRequest"
+      grant_data['operationParam'] = grant_request['operationParam']
+
+      # Channel type is mandatory in GrantData Object.
+      # Setting it to 'GAA' by default. 
+      grant_data['channelType'] = 'GAA'
+
+      # Assign grant_expire_time to grant objects.
+      grant_time_stamp = datetime.now().today() + timedelta(hours=24)
```

Comment:
Consider setting this at +2 years (or some other "really long" time) instead so that the default config will be valid for a long time (otherwise we will have to regenerate all of the time).

Reply:
Updated   accordingly